renpy/compat/pickle.py

```diff
@@ -90,6 +91,11 @@ def find_class(self, module, name):
                 elif name == "datetime":
                     return self.datetime
 
+            if module == "_ast":
```

Comment:
```suggestion
            if module == "_ast" and name in REWRITE_NODES:
                return REWRITE_NODES[name]
```
Saving a function call can be important at pickle time.

Also, if this find_class is to grow this way, maybe we should make it rely on a `{modulename : {classname : wrapper}}` dict which would make the lookup faster than sequential equality tests.

Reply:
>Saving a function call (two, even) can be important at pickle time.

Fair enough. I focussed more on the early bailout on "_ast" as most important perf item, as it should be extremely rare to enter the branch there anyway. I tried to avoid the double dict lookup in favour of the extra function call, but it seems I forgot this is python we're dealing with. That said, I don't really think it matters, as this isn't really the slowest part of the code if we hit the _ast branch.

If the amount of modules affected increases, it'll indeed be worth to move away from the sequential equality checks. I'd still keep to the `{modulename : {classname : wrapper}}` approach, as these bypasses should be rare enough that early bailout is almost always preferable.

The current approach trades an extra bytecode hit for a bit less indirection right now with the two equality checks. But regarding performance, the `_ast` path should only ever be when old-style screenlang support is hit. Nothing else in ren'py uses it, and as storing ast's is simply incredibly inefficient use of pickling to begin with, I doubt that will change. Storing source / bytecode is practically always preferable.

At which point we're already going to hit a significantly worse perf drawback with the use of the `NodeTransformer` that crawls the entire `_ast` node tree again, sadly that one was unavoidable. Normally you can coax the pickle machinery into doing some more fun things, but all ast nodes have an explicit `__reduce__` implementation that passes no arguments at construction time. Thus I could only get it to spit out a different node type unconditionally. And sadly `_ast.Name` needs to be split conditionally.

Reply:
Does it make sense to drag this to renpy 8 when you generally have to rewrite quite a bit to migrate from such an old version as renpy 6? Is there really a case where you can't recompile? Especially if you have to refactor your code for newer python/renpy. I thought this module was more of a persistent/saves compatibility thingie for the end-users.

Reply:
Beats me. The 8.0.0 release notes mention renpy 8 being used to run unmodified ren'py games from 2006, and old-style screen language is from 2014 so it felt weird to me when I discovered that it couldn't do this. What confused me mostly was that the engine had all the code ready to still support these games, except for that it wasn't reachable at all, as trying to load the rpyc file itself would fail due to some very complex interactions between python versions. I realized there was a way to work around that that wasn't too terrible so I had a go at it.

I honestly don't care much if it makes it in, but figured it'd at least be useful to offer.